app/lib/present.ts

```diff
@@ -8,31 +8,42 @@ import { Ed25519Signature2020 } from '@digitalcredentials/ed25519-signature-2020
 import type { VerifiablePresentation } from '../types/presentation';
 import type { CredentialRecordRaw } from '../model/credential';
 import type { DidRecordRaw } from '../model/did';
+import type { Credential } from '../types/credential';
 
 import { securityLoader } from './documentLoader';
 
 const documentLoader = securityLoader().build();
 
-export async function sharePresentation(rawCredentialRecords: CredentialRecordRaw[], didRecord: DidRecordRaw): Promise<void> {
-  const filePath = `${RNFS.DocumentDirectoryPath}/presentation.json`;
-  const credentials = rawCredentialRecords.map(({ credential }) => credential);
+export async function createVerifiablePresentation(
+  credentials: Credential[] = [],
+  didRecord: DidRecordRaw,
+  challenge = uuid.v4(),
```

Comment:
@dmitrizagidulin what should we add as the challenge for creating a verifiable presentation with the library? I seem to recall it being a required field.

Reply:
This method is really wrapping the create & sign presentation process and the reason the `challenge` parameter exists is to allow the deep link request to pass its own challenge instead of having one generated. I'll add a comment about that.